src/formatDate.js

```diff
@@ -8,7 +8,30 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat.pop();
+  const toSeparator = toFormat.pop();
+
+  const dateParts = date.split(fromSeparator);
+
+  const dateMap = {};
+
+  fromFormat.forEach((format, index) => {
+    dateMap[format] = dateParts[index];
+  });
+
+  if (dateMap['YY']) {
+    const year = parseInt(dateMap['YY'], 10);
+
+    dateMap['YYYY'] = year < 30 ? '20' + dateMap['YY'] : '19' + dateMap['YY'];
+  }
+
+  if (dateMap['YYYY']) {
+    dateMap['YY'] = dateMap['YYYY'].slice(-2);
```

Comment:
Similarly, this check for 'YYYY' in dateMap assumes that 'YYYY' is always followed by 'YY' in the `toFormat`. Ensure that 'YY' is actually required in the `toFormat` before slicing 'YYYY' to 'YY'.